### ISSUE TITLE
AttributeError: 'Tuple' object has no attribute 'value'

### ISSUE THREAD

**Issue by alexpetralia** [alex-sherman/deco#25]
I'm using Python 3.4 on Ubuntu.

I'm testing out `deco` in a quick program I'm writing located [here](https://github.com/alexpetralia/boston_events_advisor/blob/master/main.py). 

However, I'm getting this cryptic error:

```
runfile('/home/alexpetralia/Desktop/boston_events/main.py', wdir='/home/alexpetralia/Desktop/boston_events')
Traceback (most recent call last):

  File "<ipython-input-4-b3482a262993>", line 1, in <module>
    runfile('/home/alexpetralia/Desktop/boston_events/main.py', wdir='/home/alexpetralia/Desktop/boston_events')

  File "/usr/lib/python3/dist-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 586, in runfile
    execfile(filename, namespace)

  File "/usr/lib/python3/dist-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 48, in execfile
    exec(compile(open(filename, 'rb').read(), filename, 'exec'), namespace)

  File "/home/alexpetralia/Desktop/boston_events/main.py", line 71, in <module>
    x = populate_df()

  File "/usr/local/lib/python3.4/dist-packages/deco/conc.py", line 47, in __call__
    rewriter.visit(node.body[0])

  File "/usr/lib/python3.4/ast.py", line 245, in visit
    return visitor(node)

  File "/usr/local/lib/python3.4/dist-packages/deco/astutil.py", line 85, in visit_FunctionDef
    self.generic_visit(node)

  File "/usr/local/lib/python3.4/dist-packages/deco/astutil.py", line 55, in generic_visit
    super(NodeTransformer, self).generic_visit(node)

  File "/usr/lib/python3.4/ast.py", line 253, in generic_visit
    self.visit(item)

  File "/usr/lib/python3.4/ast.py", line 245, in visit
    return visitor(node)

  File "/usr/local/lib/python3.4/dist-packages/deco/astutil.py", line 68, in generic_visit
    name = child.targets[0].value

AttributeError: 'Tuple' object has no attribute 'value'
```

Any ideas what may be causing this?

**Comment by alex-sherman** [comment]
Cool thanks for finding this! The error is a little less cryptic in the latest version on Github which I should really get out to pip. It now says:
`ValueError: Concurrent assignment only valid for index based objects`
I should probably have that include a line number as well, but the error in your case is coming from line 29. To fix this I think you can allocate space in the `DataFrame` ahead of time and then assign values to it like:
`df[i] = scrape(i, link)`

Basically you're not allowed to keep a reference to the concurrent result, you can only tell deco where the result should go when everything is done.

Hope this helps, thanks again!

**Comment by alexpetralia** [comment]
That works! Thank you very much for your help and great job on the program!
